src/Microsoft.ML.Data/Scorers/FeatureContributionCalculatingTransformer.cs

```diff
@@ -0,0 +1,673 @@
+    "Feature Contribution Mapper", FeatureContributionCalculatingTransformer.BindableMapper.MapperLoaderSignature)]
+
+[assembly: LoadableClass(FeatureContributionCalculatingTransformer.Summary, typeof(FeatureContributionCalculatingTransformer), null, typeof(SignatureLoadModel),
+    FeatureContributionCalculatingTransformer.FriendlyName, FeatureContributionCalculatingTransformer.LoaderSignature)]
+
+[assembly: LoadableClass(typeof(IRowMapper), typeof(FeatureContributionCalculatingTransformer), null, typeof(SignatureLoadRowMapper),
+   FeatureContributionCalculatingTransformer.FriendlyName, FeatureContributionCalculatingTransformer.LoaderSignature)]
+
+namespace Microsoft.ML.Runtime.Data
+{
+    /// <summary>
+    /// The FeatureContributionCalculationTransformer scores the model on an input dataset and
+    /// computes model-specific contribution scores for each feature.
+    /// </summary>
+    /// <remarks>
+    /// See the sample below for an example of how to compute feature importance using the FeatureContributionCalculatingTransformer.
+    /// </remarks>
+    /// <example>
+    /// <format type="text/markdown">
+    /// <![CDATA[
+    /// [!code-csharp[FCT](~/../docs/samples/docs/samples/Microsoft.ML.Samples/Dynamic/FeatureContributionCalculationTransform.cs)]
+    /// ]]>
+    /// </format>
+    /// </example>
+    public sealed class FeatureContributionCalculatingTransformer : RowToRowTransformerBase
+    {
+        // Apparently, loader signature is limited in length to 24 characters.
+        internal const string Summary = "For each data point, calculates the contribution of individual features to the model prediction.";
+        internal const string FriendlyName = "Feature Contribution Transform";
+        internal const string LoaderSignature = "FeatureContribution";
+
+        private readonly string _featureColumn;
+        private readonly BindableMapper _mapper;
+
+        private static VersionInfo GetVersionInfo()
+        {
+            return new VersionInfo(
+                modelSignature: "FCC TRAN",
+                verWrittenCur: 0x00010001, // Initial
+                verReadableCur: 0x00010001,
+                verWeCanReadBack: 0x00010001,
+                loaderSignature: LoaderSignature,
+                loaderAssemblyName: typeof(FeatureContributionCalculatingTransformer).Assembly.FullName);
+        }
+
+        /// <summary>
+        /// The Feature Contribution Calculation Transform scores the model on an input dataset and
+        /// computes model-specific contribution scores for each feature.
+        /// </summary>
+        /// <param name="env">The environment to use.</param>
+        /// <param name="predictor">A trained model that supports Feature Contribution Calculation, and which will be used for scoring.</param>
```

Comment:
>A trained model [](start = 36, length = 15)

think we starting to call those Model Parameters

Reply:
For the variable name too or just the comment?

---
In reply to: [240765873](https://github.com/dotnet/machinelearning/pull/1847#discussion_r240765873) [](ancestors = 240765873)